cmake_modules/optionDefaults.cmake

```diff
@@ -2,20 +2,21 @@ if ( NOT DESTDIR )
     set( DESTDIR "" )
 endif()
 
-if ( NOT PREFIX )
-if ( WIN32 )
-    set( prog86key "ProgramFiles(x86)" )
-    if ("$ENV{${prog86key}}" STREQUAL "")
-        set( PREFIX "$ENV{ProgramFiles}" )
-    else()
-        set( PREFIX "$ENV{${prog86key}}" )
-    endif()
-    set(ORIG_PREFIX ${PREFIX})
-    string(REGEX REPLACE "\\\\" "/" PREFIX ${PREFIX})    
-else (WIN32)
-    set( PREFIX "${DESTDIR}/opt" )
-    set( ORIG_PREFIX ${PREFIX})
-endif (WIN32)
+if(NOT PREFIX)
+    if(WIN32)
+        set(prog86key "ProgramFiles(x86)")
+        if("$ENV{${prog86key}}" STREQUAL "")
+            set(PREFIX "$ENV{ProgramFiles}")
```

Comment:
I think this test is being applied on the build machine? 
IOW if the build machine is 32bit _always_ install the bundled package into the 64bit folder on 64bit machines and 32bit folder on 32bit machines.
Given we probably don't have any 32bit build machine we get away with it?

Reply:
ProgramFiles isn't really the "64 bit folder".  Its always the default unless you have a 64 bit OS and then MS adds the x86 folder for 32 bit programs.

So we just need to check if we're on a 64 bit machine, because if we're building on a 64 bit platform but then set the cmake generator to create scripts for a 32 bit build, we need to set the prefix to the x86 directory.

Reply:
But this test is for the build machine and not the install machine?  So if the build machine is 32bit and the target install machine is 64bit - the 32bit package will end up in the 64bit folder on the target install machine?

Reply:
Ah I see what you mean now.  Hrm Not sure how to make it consistent.  I have read up a little bit on Windows shell redirection and SHGetKnownFolderPath but I'm not certain how to properly implement it.

Reply:
I'll go ahead and revert any changes to the prefix, and just keep the .bat file using relative paths, since that is certain to work properly regardless of location.

Reply:
Well we know the current 32bit version resolves correctly - it should be just a matter of passing the correct flag into the NSIS installer script?

Reply:
Well the purpose of this ticket was to make it so the path in the esdl.bat always resolved correctly.  With relative paths now that has been achieved.  I think going beyond that scope and messing with the Prefix is probably a bad idea.  So my plan is to just keep the relative path changes in the esdl.bat, so long as you agree.  I need to read more into Microsoft's file system redirector.  The Prefix here is kind of scary to touch because it ends up getting embedded in so many c++ source files for different components.

Reply:
I agree - was there a different ticket for Win64 installs going to the correct folder - or did that "just work"?

Reply:
It has always "just worked" as far as I know.  Ming would probably know more about that as he has much more experience with Windows than I do.